common/lib/client/realtime.js

```diff
@@ -132,11 +132,19 @@ var Realtime = (function() {
 		return channel;
 	};
 
+	/* Included to support certain niche use-cases; most users should ignore this.
+	 * Please do not use this unless you don't know what you're doing */
 	Channels.prototype.release = function(name) {
+		name = String(name);
 		var channel = this.all[name];
-		if(channel) {
-			delete this.all[name];
+		if(!channel) {
+			return;
+		}
+		var s = channel.state;
+		if(s !== 'initialized' && s !== 'detached' && s !== 'failed') {
+			throw new ErrorInfo('Can only release a channel in a terminal state (initialized, detached, or failed); was ' + s, 90001, 400);
```

Comment:
The definitions of the words "terminal" and "initialized" don't marry up for me when used together in this context. I also can't find the word "terminal" anywhere in our [features specification](https://docs.ably.io/client-lib-development-guide/features/).

I think there could be more clarity here for the casual maintainer - perhaps even a function or 'method' that computes this conclusion (i.e. encapsulate this logic, enabling re-use and certainly providing it the formal location for documentary commentary to describe to maintainers of this codebase what it tells us and how that links to the spec.).

Reply:
Fair enough. https://github.com/ably/ably-js/pull/672/commits/c86dd0109b8b3ce317e3f27a3e833c76bb891530 any better?

re spec item: if you're happy with this, I'll write a spec item for https://github.com/ably/docs/issues/437 , and can edit it into ably-js post-facto